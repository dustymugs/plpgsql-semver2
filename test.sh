#!/bin/bash

if [ -z "$POSTGRES_SEARCH_PATH" ]; then
	POSTGRES_SEARCH_PATH="public"
fi

for sql in $(find ./tests -name test.*.sql); do
	rm -rf /tmp/semver.out
	expected=${sql/%.sql/.expected}
	echo -n $sql "..."
	psql "options=--search_path=${POSTGRES_SEARCH_PATH}" -q -A -t -f "$sql"	> /tmp/semver.out
	diffs=$(diff /tmp/semver.out "$expected")
	if [ -z "$diffs" ]; then
		echo "OK"
	else
		echo $diffs
	fi
done

rm -rf /tmp/semver.out
