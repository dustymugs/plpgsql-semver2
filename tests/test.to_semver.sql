SELECT to_semver('1.2.3');
SELECT to_semver('1.2.3-alpha');
SELECT to_semver('1.2.3-alpha.beta');
SELECT to_semver('1.2.3-alpha.beta.1');
SELECT to_semver('1.2.3-alpha.beta.1+amd64');
SELECT to_semver('1.2.3+amd64');
