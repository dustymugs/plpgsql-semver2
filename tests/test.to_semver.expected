'(1,2,3,alpha.beta.1,"[""alpha"", ""beta"", 1]",amd64)'
